Gui/opensim/build.xml

```diff
@@ -308,4 +324,76 @@
+    <target name="build-linux" depends="unzip-distro,copy-visualizer,copy-models">
+        <!-- <echoproperties/> -->
+
+        <copy todir="${opensim.dist.dir}/Resources/Code/GUI">
+            <fileset dir="Scripts" />
+        </copy>
+
+        <delete>
+            <fileset dir="${opensim.dist.dir}/OpenSim/modules/ext"
+                includes="**/jxbrowser-*mac*.jar"/>
+            <fileset dir="${opensim.dist.dir}/OpenSim/modules/ext"
+                includes="**/jxbrowser-*win*.jar"/>
+            <fileset file="${opensim.dist.dir}/bin/OpenSim.exe"/>
+            <fileset file="${opensim.dist.dir}/bin/OpenSim64.exe"/>
+            <fileset file="dist/OpenSim.zip"/>
+        </delete>
+
+        <!-- Copy the OpenSim API. -->
+        <copy todir="${opensim.dist.dir}">
+            <fileset dir="${api.dir}">
+                <!-- Don't need the java source code. -->
+                <exclude name="**/sdk/Java/org/**" />
+                <!-- Developers can build developer doxygen on their own. -->
+                <exclude name="**/html_developer/**" />
+            </fileset>
+        </copy>
+
+        <chmod file="${opensim.dist.dir}/bin/**" perm="755" />
+        <chmod file="${opensim.dist.dir}/sdk/Simbody/libexec/simbody/simbody-visualizer"
+            perm="755" />
+
+        <!-- Zip up Resources and delete the original Resources folder. -->
+        <zip basedir="${opensim.dist.dir}/Resources/" destfile="${opensim.dist.dir}/Resources.zip" />
+        <delete dir="${opensim.dist.dir}/Resources"/>
+
+        <replace file="${opensim.dist.dir}/etc/OpenSim.conf" value="${app.version}">
+            <replacefilter token="@VERSION@" />
+        </replace>
+
+        <copy tofile="${opensim.dist.dir}/LICENSE.txt"
+            file="installer_files/license_for_binary_distribution.txt"/>
+        <copy tofile="${opensim.dist.dir}/NOTICE.txt"
+            file="installer_files/notice_for_binary_distribution.txt"/>
+        <mkdir dir="${opensim.dist.dir}/third_party_license_texts"/>
+        <copy todir="${opensim.dist.dir}/third_party_license_texts/"
+            file="jfreechart/lgpl-3.0.txt"/>
+        <copy todir="${opensim.dist.dir}/third_party_license_texts/"
+            file="jfreechart/gpl-3.0.txt"/>
+        <copy todir="${opensim.dist.dir}/third_party_license_texts/"
+            file="visualViewer/src/org/opensim/topologyEditor/cddl.txt"/>
+
+        <copy todir="${opensim.dist.dir}/"
+            file="${installer-files.dir}/OpenSimLogoWhiteNoText.png" />
+        <copy todir="${opensim.dist.dir}/"
+            file="${installer-files.dir}/opensim.desktop" />
+        <copy todir="${opensim.dist.dir}/"
+            file="${installer-files.dir}/INSTALL" />
+
+        <tar destfile="${opensim.dist.dir}/../../OpenSim-${app.version}.tar.gz"
+            basedir="${opensim.dist.dir}/.."
+            excludes="OpenSim/bin/**,**/*.log,OpenSim/*.desktop,OpenSim/sdk/Simbody/libexec/simbody/simbody-visualizer,OpenSim/INSTALL"
```

Comment:
Can you explain why these files are excluded here and then included below? Just curious. Perhaps this should be documented in the file.

Reply:
The excludes/includes are to make sure everything is copied with the proper file permissions. I've since realized that these build steps (along with a number of other actions in the ant build.xml) would probably make more sense to have as part of the CMake build system, but that's an entirely different discussion.